main/src/library/BPlusTree.flix

```diff
@@ -0,0 +1,763 @@
+            let rightKeys = Array.empty(rc, order - 1);
+            let rightChildren = Array.empty(rc, order);
+            leftNode->size = mid;
+            let keyOfRight = if (insertInLeft) {
+                let keyToPass_ = Array.get(mid - 1, leftNode->keys);
+                let leftChildren = leftNode->children;
+                let leftKeys = leftNode->keys;
+                // No problems. Just copy everything from mid and forward. Finish by inserting the new element in old node.
+                Array.copyInto(srcPos = mid, dstPos = 0, len = rightSize - 1, src = leftKeys, rightKeys);
+                Array.copyInto(srcPos = mid, dstPos = 0, len = rightSize, src = leftChildren, rightChildren);
+                arrayInsert(insertedChild, insertionPoint + 1, mid, leftChildren);
+                arrayInsert(newKey, insertionPoint, mid, leftKeys);
+                leftNode->size = mid + 1;
+                keyToPass_
+            } else {
+                // Insert the new node in the new right node and copy the relevant elements of the leftNode (index>=mid).
+                let rightSizeWithoutNew = rightSize - 1;
+                if(insertionPoint + 1 == mid) {
+                    // If insertionPoint is the first element copy all old keys and old elements and insert new child at 0.
+                    Array.copyInto(srcPos = mid - 1, dstPos = 0, len = rightSizeWithoutNew, src = leftNode->keys, rightKeys);
+                    Array.copyInto(srcPos = mid, dstPos = 1, len = rightSizeWithoutNew, src = leftNode->children, rightChildren);
+                    Array.put(insertedChild, 0, rightChildren);
+                    newKey
+                } else {
+                    let childIndexInRight = insertionPoint + 1 - mid;
+                    Array.copyInto(srcPos = mid, dstPos = 0, len = childIndexInRight - 1, src = leftNode->keys, rightKeys);
+                    Array.copyInto(srcPos = mid, dstPos = 0, len = childIndexInRight, src = leftNode->children, rightChildren);
+                    Array.put(insertedChild, childIndexInRight, rightChildren);
+                    Array.put(newKey, childIndexInRight - 1, rightKeys);
+                    // If insertionPoint is the last index we are done, otherwise insert rest of list.
+                    if(childIndexInRight != rightSizeWithoutNew) {
+                        Array.copyInto(srcPos = mid + childIndexInRight - 1, dstPos = childIndexInRight, len = rightSizeWithoutNew - childIndexInRight, src = leftNode->keys, rightKeys);
+                        Array.copyInto(srcPos = mid + childIndexInRight, dstPos = childIndexInRight + 1, len = rightSizeWithoutNew - childIndexInRight, src = leftNode->children, rightChildren)
+                    } else ();
+                    Array.get(mid - 1, leftNode->keys)
+                }
+            };
+            let rightNode = new Node @ rc {
```

Comment:
Could you make two functions, mkLeaf and mkNonLeaf which only takes the non-empty arguments? easier to read and maintain.